src/otg/preprocess/finngen/summary_stats.py

```diff
@@ -0,0 +1,121 @@
+        """Summary statistics ingestion for one FinnGen study."""
+        processed_summary_stats_df = (
+            summary_stats_df
+            # Drop rows which don't have proper position.
+            .filter(f.col("pos").cast(t.IntegerType()).isNotNull()).select(
+                # Add study idenfitier.
+                f.lit(study_id).cast(t.StringType()).alias("studyId"),
+                # Add variant information.
+                f.concat_ws(
+                    "_",
+                    f.col("#chrom"),
+                    f.col("pos"),
+                    f.col("ref"),
+                    f.col("alt"),
+                ).alias("variantId"),
+                f.col("#chrom").alias("chromosome"),
+                f.col("pos").cast(t.IntegerType()).alias("position"),
+                # Parse p-value into mantissa and exponent.
+                *parse_pvalue(f.col("pval")),
+                # Add beta, standard error, and allele frequency information.
+                f.col("beta").cast("double"),
+                f.col("sebeta").cast("double").alias("standardError"),
+                f.col("af_alt").cast("float").alias("effectAlleleFrequencyFromSource"),
+            )
+            # Calculating the confidence intervals.
+            .select(
+                "*",
+                *calculate_confidence_interval(
+                    f.col("pValueMantissa"),
+                    f.col("pValueExponent"),
+                    f.col("beta"),
+                    f.col("standardError"),
+                ),
+            )
+        )
+
+        # Initializing summary statistics object:
+        return cls(
+            _df=processed_summary_stats_df,
+            _schema=cls.get_schema(),
+        )
+
+
+def ingest_finngen_summary_stats(
+    finngen_study_id,
+    finngen_summary_stats_location,
+    finngen_summary_stats_out,
+    spark_write_mode,
+):
+    """Summary stats ingestion for FinnGen.
+
+    Args:
+        finngen_study_id (str): ID of the individual study being ingested.
+        finngen_summary_stats_location (str): Google Storage URI for the summary stats to ingest.
+        finngen_summary_stats_out (str): Output path for the FinnGen summary statistics dataset.
+        spark_write_mode (str): Dataframe write mode.
+    """
+    spark = SparkSession.builder.master("yarn").appName("ingest_finngen").getOrCreate()
+
+    # Ingest the data.
+    logging.info(
+        f"Processing {finngen_study_id} with summary statistics in {finngen_summary_stats_location}"
+    )
+    summary_stats_df = (
+        spark.read.option("delimiter", "\t")
+        .csv(finngen_summary_stats_location, header=True)
+        .repartition("#chrom")
+    )
+
+    # Process and output the data.
+    out_filename = f"{finngen_summary_stats_out}"
+    FinnGenSummaryStats.from_finngen_harmonized_summary_stats(
+        summary_stats_df, finngen_study_id
+    ).df.sortWithinPartitions("position").write.partitionBy("chromosome").mode(
```

Comment:
@DSuveges did you solve your issues with this?

Reply:
Oh, yes, I would remove `partitionBy("chromosome")` from after write.

Reply:
This has been now fixed in the following way:
* The entire output is one big Parquet dataset, partitioned by study and then by chromosome.
* The individual *.parquet files will not contain either `study` or `chromosome` columns.
* However, when you ingest the entire dataset (& then filter by study and/or chromosome as required), the columns will be properly populated.